openshift/tests-extension/test/webhooks.go

```diff
@@ -0,0 +1,362 @@
+	webhookOperatorPackageName = "webhook-operator"
+	webhookOperatorCRDName     = "webhooktests.webhook.operators.coreos.io"
+)
+
+var _ = Describe("[sig-olmv1][OCPFeatureGate:NewOLMWebhookProviderOpenshiftServiceCA][Skipped:Disconnected][Serial] OLMv1 operator with webhooks",
+	Ordered, Serial, func() {
+		var (
+			k8sClient                       client.Client
+			dynamicClient                   dynamic.Interface
+			webhookOperatorInstallNamespace string
+			cleanup                         func(ctx context.Context)
+		)
+
+		BeforeEach(func(ctx SpecContext) {
+			By("initializing Kubernetes client and dynamic client")
+			k8sClient = env.Get().K8sClient
+			restCfg := env.Get().RestCfg
+			var err error
+			dynamicClient, err = dynamic.NewForConfig(restCfg)
+			Expect(err).ToNot(HaveOccurred(), "failed to create dynamic client")
+
+			By("requiring OLMv1 capability on OpenShift")
+			helpers.RequireOLMv1CapabilityOnOpenshift()
+
+			By("ensuring no ClusterExtension and CRD from a previous run")
+			helpers.EnsureCleanupClusterExtension(ctx, webhookOperatorPackageName, webhookOperatorCRDName)
+
+			By(fmt.Sprintf("checking if the %s exists", webhookCatalogName))
+			catalog := &olmv1.ClusterCatalog{}
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: webhookCatalogName}, catalog)
+			if apierrors.IsNotFound(err) {
+				By(fmt.Sprintf("creating the webhook-operator catalog with name %s", webhookCatalogName))
+				catalog = helpers.NewClusterCatalog(webhookCatalogName, "quay.io/operator-framework/webhook-operator-index:0.0.3")
+				err = k8sClient.Create(ctx, catalog)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("waiting for the webhook-operator catalog to be serving")
+				helpers.ExpectCatalogToBeServing(ctx, webhookCatalogName)
+			} else {
+				By(fmt.Sprintf("webhook-operator catalog %s already exists, skipping creation", webhookCatalogName))
+			}
+			webhookOperatorInstallNamespace = fmt.Sprintf("webhook-operator-%s", rand.String(5))
+			cleanup = setupWebhookOperator(ctx, k8sClient, webhookOperatorInstallNamespace)
+		})
+
+		AfterEach(func(ctx SpecContext) {
+			By("performing webhook operator cleanup")
+			if cleanup != nil {
+				cleanup(ctx)
+			}
+		})
+
+		It("should have a working validating webhook", func(ctx SpecContext) {
+			By("creating a webhook test resource that will be rejected by the validating webhook")
+			Eventually(func() error {
+				name := fmt.Sprintf("validating-webhook-test-%s", rand.String(5))
+				obj := newWebhookTestV1(name, webhookOperatorInstallNamespace, false)
+
+				_, err := dynamicClient.
+					Resource(webhookTestGVRV1).
+					Namespace(webhookOperatorInstallNamespace).
+					Create(ctx, obj, metav1.CreateOptions{})
+
+				switch {
+				case err == nil:
+					// Webhook not ready yet; clean up and keep polling.
+					_ = dynamicClient.Resource(webhookTestGVRV1).
+						Namespace(webhookOperatorInstallNamespace).
+						Delete(ctx, name, metav1.DeleteOptions{})
```

Comment:
why is there a deletion of the web hook happening here? 

This is within an eventually block..isn't it sufficient to 
1. Check if there is an error 
2. Check if the error contains the substring we expect 

like so: https://github.com/openshift/origin/pull/30059/files#diff-6dd6fa78ac85235012d3c9910f8510bc1640c830a83888681bd5922cec0dffcbR65-R68 ? 

The switch case block is just adding verbosity

Reply:
The switch is required to ensure the same behaviour
Before the webhook server is up and able to reject another error, we will still need to wait until we have the expected error.